### PR TITLE
golangci - enable tenv linter and swap os.SetEnv to correct t.SetEnv & enabled 

### PR DESCRIPTION
**Pull request by katbyte**
This pull request refactors the code to replace instances of os.Setenv with t.Setenv in test files. The t.Setenv method, introduced in Go 1.17, is specifically designed for managing environment variables in tests, ensuring they are properly cleaned up after each test case. currently both are in use within the provider and this aligns them with the correct methold.

attempted to enable nilerr to enforce a nolint directive & a comment explaining why the error is being ignored, however linter does not seem to respect nolint director so left comments and disabled linter

also tidy up the golangci config file

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/compute/capacity_reservation_group_resource.go [hunk 1]
```diff
@@ -172,7 +172,7 @@ func resourceCapacityReservationGroupDelete(d *pluginsdk.ResourceData, meta inte
 			Refresh: func() (interface{}, string, error) {
 				res, err := client.Delete(ctx, *id)
 				if err != nil {
-					return res, "Deleting", nil
+					return res, "Deleting", nil //nolint:nilerr  Returning nil error is intentional as we will retry the delete operation
```
Comment (jackofallops) [inline internal/services/compute/capacity_reservation_group_resource.go:175]:
```suggestion
					return res, "Deleting", nil // nolint:nilerr  Returning nil error is intentional as we will retry the delete operation
```

File: internal/services/apimanagement/migration/policy.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ func (ApiManagementPolicyV0ToV1) UpgradeFunc() pluginsdk.StateUpgraderFunc {
 	return func(ctx context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
 		apiMgmtId, err := policy.ParseServiceID(rawState["id"].(string))
 		if err != nil {
-			return rawState, nil
+			return rawState, nil //nolint:nilerr // this is not an error as we just want to skip the upgrade
```
Comment (jackofallops) [inline internal/services/apimanagement/migration/policy.go:25]:
```suggestion
			return rawState, nil // nolint:nilerr // this is not an error as we just want to skip the upgrade
```

File: .golangci.yml [hunk 3]
```diff
@@ -32,36 +32,33 @@ linters:
...
+    ##### DISABLED as it (correctly) flags fmt.Errorf("constant") to be replaced with errors.New("constant") and there are ~1500 instances of this in the codebase
     #- perfsprint # Checks that fmt.Sprintf can be replaced with a faster alternative.
 
-    #### VALID but DISABLED as we have a lot of these in the codebase to switch over
+    #### DISABLED but valid as relying on output variable names is less than idea, have a lot of these in the codebase to switch over
     #- nakedret # Checks that functions with naked returns are not longer than a maximum size
 
-    #### DISABLED TO DO IN ITS OWN PR - should be enabled and done #####
-    #- tenv #detects using os.Setenv instead of t.Setenv since Go1.17.
+    #### DISABLED as it doesn't appear to respect the no-lint directive
```
Comment (jackofallops) [inline .golangci.yml:60]: Directive should be on preceding line, and be of the form:
```
// lint:ignore nilerr reason
```
(I think)

**Review by jackofallops — CHANGES_REQUESTED** [review]
Thanks @katbyte - Just a couple of things to take a look at, and I think this will be gtg.

File: .golangci.yml (hunk 3 above)
Comment (katbyte) [reply .golangci.yml:60]: huh weird its different then the other linters but this works! thanks

File: internal/sdk/plugin_sdk_test.go [hunk 4]
```diff
@@ -624,7 +624,7 @@ func TestAccPluginSDKAndDecoderSets(t *testing.T) {
 }
 
 func TestAccPluginSDKAndEncoder(t *testing.T) {
-	os.Setenv("TF_ACC", "1")
+	os.Setenv("TF_ACC", "1") // nolint: tenv plugin sdk always is
```
Comment (katbyte) [inline internal/sdk/plugin_sdk_test.go:627]:
```suggestion
	os.Setenv("TF_ACC", "1") // lint:ignore tenv plugin sdk always is
```

**Review by mbfrahry — APPROVED** [review]
LGTM!

**Pull request merged**